src/AbstractIsoCodesConstraintValidator.php

```diff
@@ -27,10 +30,21 @@
      */
     public function validate($value, Constraint $constraint)
     {
-        $constraintClass = preg_replace('/Validator$/', '', get_class($this));
+        $validatorClass = get_class($this);
+        if (IsoCodesGenericValidator::class === $validatorClass
+            && !($constraint instanceof AbstractIsoCodesGenericConstraint || $constraint instanceof IsoCodesGeneric)
+        ) {
+            throw new UnexpectedTypeException($constraint, AbstractIsoCodesGenericConstraint::class);
+        } elseif (IsoCodesGenericValidator::class !== $validatorClass) {
+            $constraintClass = preg_replace('/Validator$/', '', $validatorClass);
 
-        if (!$constraint instanceof $constraintClass) {
-            throw new UnexpectedTypeException($constraint, $constraintClass);
+            if (!$constraint instanceof $constraintClass) {
+                throw new UnexpectedTypeException($constraint, $constraintClass);
+            }
+        }
+
+        if (!class_exists($constraint->getIsoCodesClass())) {
+            throw new ValidatorNotExistsException($constraint);
```

Comment:
In this case this is not a real problem because without `ValidatorNotExistsException`, the user will have another `ClassNotFound` exception by the system.